### ISSUE TITLE
Getting the key value using ServiceProviders

### ISSUE THREAD

**Issue by Zyles** [thephpleague/container#116]
So I may have overlooked something but I can't figure this out.

I am creating a bot that has a main object $this->bot and I want to add all the modules/services to that object.

So I setup the container etc. and I created a ServiceProvider for my module called "helloworld".

```
class ServiceProvider extends AbstractServiceProvider
{
	protected $provides = [
		'helloworld'
	];

	public function register()
	{
		$this->getContainer()->add('helloworld', 'Bot\HelloWorld\HelloCommands');
	}
}
```
I then run the __construct of the main Bot class to register and load every single service I have:

```
class Bot {

	/**
	 * Construct the bot and read the configurations.
	 */
	public function __construct()
	{
		// Create the main DependencyInjection
		$bot = new Container;

		// Read the config into the main bot object
		$this->bot->config = $this->config();

		// Add services to the container from the config
		foreach ($this->bot->config['services'] as $service) {
			$bot->addServiceProvider($service);
		}

		$this->bot = $bot;
	}

        public function config()
	{
		return [
			"services" => [
				"Bot\HelloWorld\ServiceProvider"
			],
		];
	}
}
```

This is where I get stuck. How do I access the key "helloworld" when I have registered the service as "Bot\HelloWorld\ServiceProvider" so I can create this object:

```
$this->bot->helloworld
# So I can use the commands in HelloCommands class:
$this->bot->helloworld->sayHello();
```

Basically automate the following without knowing the namespace/key:
```
$this->bot->helloworld = $bot->get('helloworld');
```

Thanks.

**Comment by philipobenito** [comment]
`$this->bot->get('helloworld')->sayHello();`

**Comment by philipobenito** [comment]
To access any key from the container, pass the name of that key to the `get` method and it will return what is stored against that key.

**Comment by Zyles** [comment]
I updated my most. I mean getting the key without knowing the key name.

Your example is manual work, I want to automate it.

**Comment by philipobenito** [comment]
That's beyond the scope of the container, you'd have to write a magic `__get` method or maybe use `ArrayAccess` to achieve something like that.

**Comment by Zyles** [comment]
Okay that is too bad. I can see there is a providers with key value in the object. Would have been most helpful if this would be possible to reverse it.

I will try and figure out how to do this another way then.

**Comment by philipobenito** [comment]
It can only give you a manual way of retrieval. To automate that would have to be up to you though as it's a very specific use case
